code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/Optimize.java

```diff
@@ -174,24 +174,46 @@ public void call(final Event event) {
                             }
 
                             if (eventData.containsKey(
+                                    OptimizeConstants.EventDataKeys.RESPONSE_ERROR)) {
+                                Object error =
+                                        eventData.get(
+                                                OptimizeConstants.EventDataKeys.RESPONSE_ERROR);
+                                if (error instanceof Map) {
+                                    failWithOptimizeError(
+                                            callback,
+                                            AEPOptimizeError.toAEPOptimizeError(
+                                                    (Map<String, ? extends Object>) error));
+                                }
+                            }
+
+                            if (!eventData.containsKey(
                                     OptimizeConstants.EventDataKeys.PROPOSITIONS)) {
-                                final Map<DecisionScope, OptimizeProposition> propositionsMap =
-                                        new HashMap<>();
-                                final Map<String, Object> propositionData =
-                                        DataReader.getTypedMap(
-                                                Object.class,
-                                                eventData,
-                                                OptimizeConstants.EventDataKeys.PROPOSITIONS);
-
-                                final OptimizeProposition optimizeProposition =
-                                        OptimizeProposition.fromEventData(propositionData);
-                                if (optimizeProposition != null
-                                        && !OptimizeUtils.isNullOrEmpty(
-                                                optimizeProposition.getScope())) {
-                                    final DecisionScope scope =
-                                            new DecisionScope(optimizeProposition.getScope());
-                                    propositionsMap.put(scope, optimizeProposition);
+                                return;
```

Comment:
This logic doesn't seem to align with iOS where both the errors and propositions (even if they are empty) is returned. Could you please verify?